Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/TransformComponent.cpp

```diff
@@ -934,57 +934,66 @@ namespace AzToolsFramework
                 return AZ::Failure(AZStd::string("Trying to set an entity ID to something that isn't an entity ID."));
             }
 
-            AZ::EntityId actualValue = static_cast<AZ::EntityId>(*((AZ::EntityId*)newValue));
+            AZ::EntityId newParentId = static_cast<AZ::EntityId>(*((AZ::EntityId*)newValue));
 
-            if (!actualValue.IsValid())
+            if (!newParentId.IsValid())
             {
                 // Handled by the calling code.
                 return AZ::Success();
             }
 
-            // Only allow reparenting the selected entities if they are all under the same instance.
-            // We check the parent entity separately because it may be a container entity and
-            // container entities consider their owning instance to be the parent instance
-            EntityIdList entitiesToCheck;
-            entitiesToCheck.reserve(2);
-            entitiesToCheck.push_back(actualValue);
-            entitiesToCheck.push_back(GetEntityId());
-            const Prefab::PrefabPublicInterface* prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get();
-            if (prefabPublicInterface && !prefabPublicInterface->EntitiesBelongToSameInstance(entitiesToCheck))
-            {
-                return AZ::Failure(AZStd::string("An entity in one prefab cannot be a child to an entity in another prefab"));
-            }
+            AZ::EntityId selectedEntityId = GetEntityId();
 
             // Prevent setting the parent to the entity itself.
-            if (actualValue == GetEntityId())
+            if (newParentId == selectedEntityId)
             {
                 return AZ::Failure(AZStd::string("You cannot set an entity's parent to itself."));
             }
 
+            // Prevent parenting to a different owning prefab instance.
+            // Note: The logic should be the same as defined in EntityOutlinerListModel::CanReparentEntities()
```

Comment:
If it's the same code, then maybe it can be moved to a utils file like EditorEntityHelpers.cpp ?

Reply:
Yeah I will look into that. It makes sense to gather the same code there.